src/main/java/seedu/address/model/person/PersonMatchesKeywordsPredicate.java

```diff
@@ -0,0 +1,100 @@
+package seedu.address.model.person;
+
+import seedu.address.commons.util.StringUtil;
+
+import java.util.function.Predicate;
+import java.util.Arrays;
+
+
+public class PersonMatchesKeywordsPredicate implements Predicate<Person> {
+    private final String keywords;
+
+    public PersonMatchesKeywordsPredicate(String keywords) {
+        this.keywords = keywords;
+    }
+
+    static int getLevenshteindist(String str1, String str2) {
+
+        int[][] arr = new int[str1.length() + 1][str2.length() + 1];
+
+        for (int i = 0; i <= str1.length(); i++)
+        {
+            for (int j = 0; j <= str2.length(); j++) {
+                if (i == 0) {
+                    arr[i][j] = j;
+                }
+
+                else if (j == 0) {
+                    arr[i][j] = i;
+                }
+
+                else {
+                    arr[i][j] = minEdits(arr[i - 1][j - 1]
+                                    + numOfReplacement(str1.charAt(i - 1),str2.charAt(j - 1)),
+                            arr[i - 1][j] + 1,
+                            arr[i][j - 1] + 1);
+                }
+            }
+        }
+        return arr[str1.length()][str2.length()];
+    }
```

Comment:
@sh4nH Could you add a concise comments to document this part of code. Otherwise it might be confusing to others. It's the best if you can add reference here to help teammates understand where does this algorithm comes from.

Also, please add the `access modifier` `private`.